### PR TITLE
Create an Embedded Java-Python app that uses a proxied Python class

### PR DESCRIPTION
**Pull request by bhoran**
I'm sure the Java and Python code could be improved, so feel free to make suggestions. 
However, this demo now runs on the JVM, so I'd like to get it approved before the end of July 2024.

### CODE REVIEW

File: graalpy-embedded-pillow/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Using GraalPy to create an Instance of a Proxied Python Class from a Java Application
```
Comment (olyagpl) [inline graalpy-embedded-pillow/README.md:1]: Let's use a simpler name for a demo:
```markdown
# Image Manipulation Java Application with Python Pillow Library Embedded

This application demonstrates how to use GraalPy to create an instance of a proxied Python class from a Java application, built with Maven.
```

Comment (olyagpl) [reply graalpy-embedded-pillow/README.md:1]: The other variant for a title:
```
# Image Manipulation Demo Using Java and GraalPy
```

File: graalpy-embedded-pillow/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Using GraalPy to create an Instance of a Proxied Python Class from a Java Application 
+
+This small application demonstrates how to use GraalPy to create an instance of a proxied Python class from a Java application, built with Maven.
+
+It includes the following steps:
+1. Install the [pillow](https://pillow.readthedocs.io/en/stable/) and [requests](https://pypi.org/project/requests/) modules. (See the [pom.xml](./pom.xml) file for more details.)
+2. Create a Python class ([pillowImageWrapper.py](src/main/resources/vfs/proj/pillowImageWrapper.py)) creates, manipulates, and stores an image.
```
Comment (olyagpl) [inline graalpy-embedded-pillow/README.md:7]:
```suggestion
2. Create a Python class ([pillowImageWrapper.py](src/main/resources/vfs/proj/pillowImageWrapper.py)) that creates, manipulates, and stores an image.
```

File: graalpy-embedded-pillow/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+3. Create a Java interface ([PillowImageProxy.java](src/main/java/com/oracle/example/graalpy/PillowImageProxy.java)) that acts as a proxy for the wrapper.
+4. Create a Swing JFrame ([PillowFrame.java](src/main/java/com/oracle/example/graalpy/PillowFrame.java)) that calls the proxy's methods. (This class also contains a `main()` method.)
+
+## Preparation
+
+Install a JDK 21, ensure that it is on your classpath, and set the value of `JAVA_HOME` accordingly.
```
Comment (olyagpl) [inline graalpy-embedded-pillow/README.md:13]: I wouldn't give a JDK version here. It'll be JDK 25 soon. 
```suggestion
Install a JDK, ensure that it is on your classpath, and set the value of `JAVA_HOME` accordingly.
```

File: graalpy-embedded-pillow/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+We recommend GraalVM for JDK 21 that can be easily installed on macOS and Linux using [SDKMAN!](https://sdkman.io/). (For other download options, see [GraalVM Downloads](https://www.graalvm.org/downloads/).)
+```bash
+sdk install java 21.0.2-graal
```
Comment (olyagpl) [inline graalpy-embedded-pillow/README.md:16]: Does it work with JDK 23 EA build?

File: graalpy-embedded-pillow/pom.xml [hunk 5]
```diff
@@ -0,0 +1,177 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (olyagpl) [inline graalpy-embedded-pillow/pom.xml:2]: We don't usually add the copyright header in Gradle/Maven config files.

File: graalpy-embedded-pillow/pom.xml [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.oracle.example</groupId>
+  <artifactId>embedded-pillow</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <!-- Select the GraalPy version to use. -->
+    <graalpy.version>24.0.1</graalpy.version>
+    <graalpy-plugin.version>24.0.0</graalpy-plugin.version>
```
Comment (olyagpl) [inline graalpy-embedded-pillow/pom.xml:54]: Should the plugin version match graalpy? 24.0.0 > 24.0.1

File: graalpy-embedded-pillow/pom.xml [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        <executions>
+          <execution>
+            <configuration>
+              <!-- specify python packages as if used with pip -->
+              <packages>
+                <package>pillow</package>
+                <package>requests</package>
+              </packages>
+              <pythonHome>
+                <!-- java-like regular expression what file paths should be included into venv/home;
```
Comment (olyagpl) [inline graalpy-embedded-pillow/pom.xml:101]: java-like  > Java-like

File: graalpy-embedded-pillow/pom.xml [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+      <id>native</id>
+      <properties>
+        <nativeimage.agentlib.argument>-agentlib:native-image-agent=config-merge-dir=${project.build.outputDirectory}/META-INF/native-image/</nativeimage.agentlib.argument>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.graalvm.buildtools</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>0.9.27</version>
```
Comment (olyagpl) [inline graalpy-embedded-pillow/pom.xml:154]: 0.9.27 > 0.10.2

**Review by olyagpl — CHANGES_REQUESTED** [review]

File: graalpy-embedded-pillow/README.md (hunk 1 above)
Comment (bhoran) [reply graalpy-embedded-pillow/README.md:1]: How about "Image Manipulation using a Java Application with an Embedded Python Library"

File: graalpy-embedded-pillow/pom.xml (hunk 6 above)
Comment (bhoran) [reply graalpy-embedded-pillow/pom.xml:54]: Apparently not, this is taken from Tim's example.

File: graalpy-embedded-pillow/README.md (hunk 1 above)
Comment (olyagpl) [reply graalpy-embedded-pillow/README.md:1]: using > Using

File: graalpy-embedded-pillow/pom.xml (hunk 8 above)
Comment (bhoran) [reply graalpy-embedded-pillow/pom.xml:154]: I'm reluctant to change this in case it breaks!

**Comment by olyagpl** [comment]
@bhoran, do you mind adding an entry to the top-level README.md? This way we can track easily if the GitHub workflow passed.

File: graalpy-embedded-pillow/pom.xml [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
```
Comment (olyagpl) [inline graalpy-embedded-pillow/pom.xml:155]:
```suggestion
<version>0.10.2</version>
```

File: .github/workflows/graalpy-embedded-pillow.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  run:
+    name: Run 'graalpy-embedded-pillow'
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v4
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '21.0.2'
```
Comment (olyagpl) [inline .github/workflows/graalpy-embedded-pillow.yml:23]:
```suggestion
java-version: '21'
```

**Review by olyagpl — CHANGES_REQUESTED** [review]

**Review by olyagpl — APPROVED** [review]

**Pull request merged**